### PR TITLE
Addresses #332: pass config-file path for roadrunner

### PR DESCRIPTION
**Pull request by kevincobain2000**
Submitted with intentions for a first review and consideration.

Reasons for this pull request:

1. As mentioned in #332: to be able use different ``.rr.yaml`` based on different environments
2. We use ``deployer``. As this is symbolic link deployment, the ``base_path()`` picks up the path from the ``releases/<id>`` reference instead of the symb link. It'd be nice if we could provide a path via cli.

The first option is more encouraging, as this config file is most likely to be on the server in a different location.

### CODE REVIEW

File: src/Commands/StartRoadRunnerCommand.php [hunk 1]
```diff
@@ -120,6 +119,20 @@ protected function writeServerStateFile(
...
+     *
+     * @return string
+     */
+    protected function configPath()
+    {
+        if (! $this->option('config-path')) {
+            touch(base_path('.rr.yaml'));
+            return base_path('.rr.yaml');
+        }
+        return $this->option('config-path');
```
Comment (Namoshek) [inline src/Commands/StartRoadRunnerCommand.php:133]: Would it make sense here to allow different styles of config paths to be passed?
```
php artisan octane:start --config-path=rr.prod.yml
php artisan octane:start --config-path=config/rr.prod.yml
php artisan octane:start --config-path=./config/rr.prod.yml
php artisan octane:start --config-path=../config/rr.prod.yml
php artisan octane:start --config-path=/etc/roadrunner/rr.prod.yml
```

Comment (kevincobain2000) [inline src/Commands/StartRoadRunnerCommand.php:133]: Something like below you mean?

```suggestion
        $configPath =  $this->option('config-path');
        if (! $this->isAbsolutePath($configPath)) {
            return base_path($configPath);
        }
        return $configPath;
```

And have a new function isAbsolutePath()

```
>>> function isAbsolutePath($path) {
...     if($path === null || $path === '') throw new Exception("Empty path");
...     return $path[0] === DIRECTORY_SEPARATOR || preg_match('~\A[A-Z]:(?![^/\\\\])~i',$path) > 0;
... }
```

File: src/Commands/StartRoadRunnerCommand.php [hunk 2]
```diff
@@ -120,6 +119,20 @@ protected function writeServerStateFile(
...
+    }
+
```
Comment (kevincobain2000) [inline src/Commands/StartRoadRunnerCommand.php:135]:
```suggestion

private function isAbsolutePath($path) {
    if($path === null || $path === '') throw new Exception("Empty path");
    return $path[0] === DIRECTORY_SEPARATOR || preg_match('~\A[A-Z]:(?![^/\\\\])~i',$path) > 0;
}
```

File: src/Commands/StartRoadRunnerCommand.php (hunk 1 above)
Comment (kevincobain2000) [reply src/Commands/StartRoadRunnerCommand.php]: Thank you, I have added absolute path 58f1dcf

**Comment by driesvints** [comment]
The flag was renamed to `--rr-config`.

**Pull request merged**